### PR TITLE
add devcontainer setup

### PR DESCRIPTION
**Pull request by bkleiner**
- add support for the [vscode devcontainer integration](https://code.visualstudio.com/docs/devcontainers/containers)
- extend `setup.sh` to take care of the bootstrap process
- update readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,22 @@
...
-~/hdz_goggle/code$ cat media.tar.gzaa media.tar.gzab > media.tar.gz
-~/hdz_goggle/code$ tar xzvf media.tar.gz
-~/hdz_goggle/code$ tar xzvf toolchain.tar.gz
-```
+This repository supports the [vscode devcontainer](https://code.visualstudio.com/docs/devcontainers/containers) integration.  
+To get started install docker, vscode and the decontainer extension.  
+A [prompt](https://code.visualstudio.com/docs/devcontainers/create-dev-container#_add-configuration-files-to-a-repository) to reopen this repository in a container should appear.
 
-## 2. Setup cmake environment
+## Building locally
```
Comment (razzeee) [inline README.md:9]: I find this a bit confusing, as this is only a setup step and not actually building anything, so making that clear would be helpful. So the old naming was probably better here.

Comment (bkleiner) [reply README.md:9]: ...i hear you, what do you suggest? how about "Setup local environment"? i want to make sure that it is conwayed that is a either or deal.

**Comment by razzeee** [comment]
Building seems to fail for me with `/bin/sh: 1: ./mkapp_ota: Permission denied` which might be fixed by one of the other PRs setting executeable flags?

It also seems to complain a lot about 
```
arm-openwrt-linux-muslgnueabi-gcc.bin: warning: environment variable 'STAGING_DIR' not defined
```

**Comment by bkleiner** [comment]
> Building seems to fail for me with `/bin/sh: 1: ./mkapp_ota: Permission denied` which might be fixed by one of the other PRs setting executeable flags?
yup https://github.com/hd-zero/hdzero-goggle/pull/3

> It also seems to complain a lot about
> 
> ```
> arm-openwrt-linux-muslgnueabi-gcc.bin: warning: environment variable 'STAGING_DIR' not defined
> ```
yup...that toolchain is not exactly optimal (warning is caused by a wrapper intended for the openwrt buildroot), but baby steps :)

**Comment by razzeee** [comment]
New readme is lovely.

Very nitpick, me problem,
I don't get why people do this cmd prefixing:

```
~/hdzero-goggle$ 
```

It only breaks copy and paste, is that what it's for? :)

**Comment by bkleiner** [comment]
> New readme is lovely.
> 
> Very nitpick, me problem, I don't get why people do this cmd prefixing:
> 
> ```
> ~/hdzero-goggle$ 
> ```
> 
> It only breaks copy and paste, is that what it's for? :)

yeah i am not a huge fan either, but i guess the idea is to make directory changes more visible.
every line explicitly shows you what CWD is expected.

**Comment by ligenxxxx** [comment]
I modify setup.sh file format from dos to unix, then setup.sh can run in windows docker.

**Comment by razzeee** [comment]
If it runs in docker, the unix format should work, as were already in the Debian container.

You probably meant for the local setup?

**Comment by bkleiner** [comment]
> If it runs in docker, the unix format should work, as were already in the Debian container.
> 
> You probably meant for the local setup?

the issue seems to be that windows still defaults to checking out the repository with CRLF.
this then causes issues when the scripts are mounted into the container. 
so i introduced the gitattributes file, trying to force the it to LF, but i guess it doesn't change the line-endings on existing checkouts.

**Comment by ligenxxxx** [comment]
Before we merge this PR, we should also test it on the ubuntu system of the virtual machine.

**Comment by razzeee** [comment]
Just tried again and it seems to build fine for me on Manjaro Linux, using the devcontainer.

**Pull request merged**